### PR TITLE
fix: Port userStore to TypeScript

### PR DESCRIPTION
**Pull request by HoonBaek**
### Description Of Changes

* Port files under `src/lib/user` to TypeScript

### CODE REVIEW

File: src/lib/dux/sdk/thunks.ts [hunk 1]
```diff
@@ -5,19 +5,18 @@ import { OpenChannelModule } from '@sendbird/chat/openChannel';
 import { GroupChannelModule } from '@sendbird/chat/groupChannel';
 
 import { SDK_ACTIONS, SdkActionTypes } from './actionTypes';
-/* eslint-disable-next-line import/named */
-import userStoreActionTypes, { UserStoreActionTypes } from '../user/actionTypes';
```
Comment (HoonBaek) [inline src/lib/dux/sdk/thunks.ts:9]: `../user/actionTypes` is being imported like a TypeScript file

File: src/lib/dux/user/actionTypes.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { CreateAction } from '../../../utils/typeHelpers/reducers/CreateAction';
+
+export const INIT_USER = 'INIT_USER';
+export const RESET_USER = 'RESET_USER';
+export const SET_USER_LOADING = 'SET_USER_LOADING';
+export const UPDATE_USER_INFO = 'UPDATE_USER_INFO';
```
Comment (sravan-s) [inline src/lib/dux/user/actionTypes.ts:6]: lets remove this and use `USER_ACTIONS .xxx`

File: src/lib/dux/user/reducers.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { UserActionTypes, USER_ACTIONS } from './actionTypes';
+import initialState, { UserStoreStateType } from './initialState';
+
+export default function reducer(state: UserStoreStateType, action: UserActionTypes): UserStoreStateType {
+  switch (action.type) {
```
Comment (sravan-s) [inline src/lib/dux/user/reducers.ts:5]: ps. can I ask you to use ts-pattern if its not too much? https://github.com/sendbird/sendbird-uikit-react/blob/main/src/lib/dux/sdk/reducers.ts

File: src/lib/dux/sdk/thunks.ts [hunk 4]
```diff
@@ -5,19 +5,18 @@ import { OpenChannelModule } from '@sendbird/chat/openChannel';
...
 const { INIT_SDK, SET_SDK_LOADING, RESET_SDK, SDK_ERROR } = SDK_ACTIONS;
-const { INIT_USER, UPDATE_USER_INFO, RESET_USER } = userStoreActionTypes;
+const { INIT_USER, UPDATE_USER_INFO, RESET_USER } = USER_ACTIONS;
 const APP_VERSION_STRING = '__uikit_app_version__';
 const IS_ROLLUP = '__is_rollup__';
 const IS_ROLLUP_REPLACE = '__is_rollup_replace__';
 
 type SdkDispatcher = React.Dispatch<SdkActionTypes>;
-type UserDispatcher = React.Dispatch<{ type: UserStoreActionTypes, payload?: any }>;
+type UserDispatcher = React.Dispatch<{ type: UserActionTypes, payload?: any }>;
```
Comment (sravan-s) [inline src/lib/dux/sdk/thunks.ts:19]: type UserDispatcher = React.Dispatch<type: UserActionTypes>

File: src/lib/dux/user/actionTypes.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  RESET_USER: 'RESET_USER',
+  SET_USER_LOADING: 'SET_USER_LOADING',
+  UPDATE_USER_INFO: 'UPDATE_USER_INFO',
+} as const;
+
+type USER_PAYLOAD_TYPES = {
+  [USER_ACTIONS.INIT_USER]: any,
+  [USER_ACTIONS.RESET_USER]: any,
+  [USER_ACTIONS.SET_USER_LOADING]: any,
+  [USER_ACTIONS.UPDATE_USER_INFO]: any,
```
Comment (sravan-s) [inline src/lib/dux/user/actionTypes.ts:19]: try using actual types

File: src/lib/dux/user/actionTypes.ts (hunk 2 above)
Comment (HoonBaek) [reply src/lib/dux/user/actionTypes.ts:6]: Oh I missed to remove them, thank you

File: src/lib/dux/user/reducers.ts (hunk 3 above)
Comment (HoonBaek) [reply src/lib/dux/user/reducers.ts]: yea sure

File: src/lib/dux/user/actionTypes.ts (hunk 5 above)
Comment (sravan-s) [reply src/lib/dux/user/actionTypes.ts:19]:
```
  [USER_ACTIONS.INIT_USER]: User,
  [USER_ACTIONS.RESET_USER]: null,
  [USER_ACTIONS.SET_USER_LOADING]: null,
  [USER_ACTIONS.UPDATE_USER_INFO]: User,
  ```

File: src/lib/dux/sdk/reducers.ts [hunk 6]
```diff
@@ -32,5 +32,5 @@ export default function reducer(state: SdkStoreStateType, action: SdkActionTypes
     })
     .otherwise(() => {
       return state;
-    });
+    }) as SdkStoreStateType;
```
Comment (sravan-s) [inline src/lib/dux/sdk/reducers.ts:35]: this worked fine before hmmmm

**Review by AhyoungRyu — APPROVED** [review]

**Review by sravan-s — APPROVED** [review]

**Pull request merged**